Dev branch 2

knappett opened the pull request:
I've added a test for the case where there's a newer GWS directory than in the archive. The test shows that symlark reports that the GWS correctly contains a symlink to the existing archive version, but I don't think that symlark highlights that a newer version exists in the GWS. Let's discuss next week!

Also, I had to add in a check for the current working directory (and change to basedir if not already there) in setup_container_dir because the 'arc_links' if statement was assuming that a change of directory to basedir had already occurred in the 'latest' if statement.

agstephens reviewed (changes requested):
Hi @knappett, Looking at lines 52-55. Maybe we should just put `os.chdir(basedir)` before line 47 (`if latest:`). If we do that then we can delete your new lines and my line 48. Might be more efficient and achieves the same result.

knappett commented:
Hi Ag, following our discussion this morning I added in the caplog.record assertions to the test_newer_gws_than_archive function in test_all.py and found that some of the paths that we had added to output messages in symlark were incorrect. I've now corrected the paths in symlark so that they point to the correct directories and the pytest run (pytest -vvv -k newer --pdb) passes all 6 tests.

agstephens commented:
Great work @knappett !
